js/src/evaluation/llm_evaluator.ts

```diff
@@ -0,0 +1,248 @@
+import { initChatModel } from "langchain/chat_models/universal";
+import { ChatPromptTemplate } from "@langchain/core/prompts";
+import * as uuid from "uuid";
+import { EvaluationResult, EvaluationResults, RunEvaluator } from "./evaluator.js";
+import type { Run, Example } from "../schemas.js";
+export class CategoricalScoreConfig {
+  key: string;
+  choices: string[];
+  description: string;
+  includeExplanation: boolean;
+  explanationDescription?: string;
+
+  constructor(params: {
+    key: string;
+    choices: string[];
+    description: string;
+    includeExplanation?: boolean;
+    explanationDescription?: string;
+  }) {
+    this.key = params.key;
+    this.choices = params.choices;
+    this.description = params.description;
+    this.includeExplanation = params.includeExplanation ?? false;
+    this.explanationDescription = params.explanationDescription;
+  }
+}
+  
+export class ContinuousScoreConfig {
+  key: string;
+  min: number;
+  max: number;
+  description: string;
+  includeExplanation: boolean;
+  explanationDescription?: string;
+
+  constructor(params: {
+    key: string;
+    description: string;
+    min?: number;
+    max?: number;
+    includeExplanation?: boolean;
+    explanationDescription?: string;
+  }) {
+    this.key = params.key;
+    this.min = params.min ?? 0;
+    this.max = params.max ?? 1;
+    this.description = params.description;
+    this.includeExplanation = params.includeExplanation ?? false;
+    this.explanationDescription = params.explanationDescription;
+  }
+}
+
+type ScoreConfig = CategoricalScoreConfig | ContinuousScoreConfig;
+
+function createScoreJsonSchema(
+  scoreConfig: ScoreConfig,
+  reasoningKey: string
+): Record<string, any> {
+  const properties: Record<string, any> = {};
+
+  if (scoreConfig.includeExplanation) {
+    properties[reasoningKey] = {
+      type: "string",
+      description: scoreConfig.explanationDescription || "The explanation for the score.",
+    };
+  }
+
+  if ("choices" in scoreConfig) {
+    properties.score = {
+      type: "string",
+      enum: scoreConfig.choices,
+      description: `The score for the evaluation, one of ${scoreConfig.choices.join(", ")}.`,
```

Comment:
This is really confusing to me tbh. 

Why do we put categorical in the numeric score field? Why call it a score? Why not call it "category" or something obvious?

Score implies sortability / comparability to me

Reply:
I changed the key to "value", but we also do this in the output parsing which I think solves this a little? not 100% sure, lmk what you think:
```
if ("choices" in this.scoreConfig) {
      const value = output.score;
      const explanation = output[this.reasoningKey];
      return {
        key: this.scoreConfig.key,
        value,
        comment: explanation,
        sourceRunId: output.runId
      };
    }
```